src/Core/Grid/Data/Factory/CustomerDiscountGridDataFactory.php

```diff
@@ -72,11 +72,19 @@ public function getData(SearchCriteriaInterface $searchCriteria)
             false
         );
 
+        $numDiscounts = count($discounts);
+
+        $discounts = array_slice(
```

Comment:
This is something that must be done in `getCustomerCartRules` logic instead here :thinking: 
If we have thousands of customer cart rules, we will load them all, then we do an array_slice, it's really reall ybad for performance :(

Reply:
I understand but I think it's not the place to refactor the `getCustomerCartRules` logic which is really complex.

Reply:
@PierreRambaud maybe let's accept it now, and we can improve it later, I think we need a better way of getting CartRule because there's a lot of method doing almost the same thing, maybe we could introduce some `CartRuleFinder`

The entire Customer view page needs a lot of work to optimize it, right now it's a mess

Reply:
ping @matks and @jolelievre need your feedback here.
Personally, I'm not in favor of merging at is :/

Reply:
> remove the false pagination completely if it's not functional, this way if it's a problem for users they will give us honest feedback about a true missing feature, not one that was hidden under the carpet 😅

In that case, we would have a long list of discounts after recent changes in the specs. I've tried to talk with PMs about changing them but without success. 

> (I'm saying this because I know, as you do 😛 , when we say "let's do this later" we forget it anyway ^^)

That's true, and given that we have specs for this section, maybe it would be good to rework it now.

I have a very similar PR: https://github.com/PrestaShop/PrestaShop/pull/24797. So these two have to be combined IMHO.

Reply:
@kpodemski I agree both PR look similar in their goal, I guess they could be merged indeed And since in your PR you introduce a new function I think it should be possible to integrate pagination in the request without risking any side effects somewhere else since the function will only be used here

But the changes will need to target 1.7.8.1